\section{Classification of Regulatory Regions}
%todo: ampliare questa sezione anche con eventuali riferimenti
The aim of this work is to test whether multi-task deep learning (Section~\ref{sec:MTLsection}) performs better than current state-of-art
machine learning approaches (in particular single task neural networks, Section~\ref{sec:singletaskNN}) in the classification of DNA cis-regulatory regions (CRRs). 

The development and functioning of the different types of cells lines of the human body require a precise spatial-temporal expression of the thousand of genes in the genome. The cell achieves this fundamental task by regulating the rate of transcription of its genes. This mechanism is often called transcriptions regulation. 
RNA polymerase II enzyme produce the transcript of a protein-coding gene, originated from the transcription start site (TSS). This enzyme needs other elements, called general transcription factors (GTF), to recognize the TSS. In this context, the main players involved could be categorized into two groups: trans-acting factors and cis-acting elements (CRR). The former is not part of DNA and can operate as gene activators or as gene repressors. The latter instead are regions along DNA \cite{NarlikarRegulaotryelements}. In this works, we primarily focus on the second group: the cis-regulatory regions.

Basically, CRRs are non-coding DNA sections which
regulate the transcription of the genes. 
DNA cis-regulatory regions could be divided into promoters, enhancers, silencers and insulators. These elements cooperate to govern a coordinated expression pattern of the gene.  
\emph{Promoters} are 100â€“1000 base pairs (bp) long DNA sequences
usually located very close to the beginning of the transcription initiation site. They define where the RNA polymerase should start the transcription of a gene. 
\emph{Enhancers} are 50 to 1500 bp DNA sequences that, when bound to special proteins (Transcription Factors or TF), augment the transcription of the associated gene. Enhancers are often found at a greater distance from TSS respect promoters.
\emph{Silencers} are regulatory regions that act as repressors on the target gene. They could be present within an enhancer or independently. 
Finally, \emph{insulators} are regions that stop the spreading of activating or repressing transcriptional activity from a locus to another. In this works, the attention is focused on distinguishing enhancers and promoters \cite{NarlikarRegulaotryelements}. 

Despite the fact that mutations in regulatory regions are considered less likely to have a phenotypic impact, recently, the number of recorded cases of non-coding mutations linked to human diseases is increased.
Thus, identifying these regions across
human DNA is became a fundamental step to understand the impact of
genetic variation on phenotype. 
Moreover, compared to coding regions, non-coding regulatory regions have few distinguishing sequence features, which makes them harder to spot. As a result, the list of regulatory elements is far to be comprehensive \cite{NarlikarRegulaotryelements}. 

The development of new computational and
biological technologies gives researchers an impressive amount of new data and therefore the possibility of developing new computational methods
for CRRs identification.



\section{Previous Works}
% TODO: RILEGGERE
% TODO: scrivere dove spiego supervised neural network single task e dove spiego gli algoritmi multi task learning.
With the rise of \emph{next-generation sequences} (NGS) techniques, which are
capable of identifying various aspects of genomes, we have seen the generation of an incredible amount of biological information. Specialized consortia usually collect this data. Among these, one can mention: the ENCODE
(Encyclopedia of DNA Elements) Consortium \cite{ENCODE_data}, which aims at building a list of functional elements in the human genome; Roadmap Epigenomics Program \cite{ROADMAP} with the aim of creating an epigenomic atlas for primary cells and tissues in human and finally FANTOM5 Project \cite{FANTOM_data} which is trying to uncover transcriptional regulatory networks based on transcript initiation positions.

This volume of NGS data has allowed the development of new computational
methods with a particular focus on both unsupervised and supervised machine
learning approaches. An extensive review of these techniques was compiled by Li et al. \cite{LiMLReview}. The first work in this direction, motivated by small sets of reliable annotations, used unsupervised methods. Among these, chromHMM \cite{ernst2012chromhmm}
and Segway \cite{HoffmanSegway} that use respectively Hidden Markov Models
(HMM) and Dynamic Bayesian Network (DBN) to discover chromatin states are the most important contributions. Given the low accuracy of the above mentioned unsupervised models, the researchers focused their attention on using additional experimental features and
different machine learning approaches. In this context, many supervised
methods such as Random Forest \cite{RFECS}, AdaBoost-based model
\cite{DELTA_adaboost} and multinomial logistic regression with LASSO
regularization \cite{ChenMultinomialLASSORegression} have attempted to predict enhancers regions inside DNA. The emergence of new
laboratory methods made it possible for FANTOM5 Consortium to identify an
atlas of transcriptionally active promoters and a set of transcriptionally
active enhancers. In \cite{DEEP}, the possibility to distinguish enhancers by
training a support vector machine with this data is put forward.

The advancement of deep learning made it possible to capture
high-level knowledge from data. This new approach has improved
the state-of-the-art of many research areas, such as speech recognition and
computer
vision \cite{lecun2015deeplearning}. Bioinformaticians have
applied deep learning techniques in many ways, ranging from predicting the impact
of variations on exon splicing \cite{XiongExon} to predicting proteins
secondary structure \cite{SecondaryStructure}. Deep learning was beneficially used in the context of cis-regulatory elements classifications. 
Deep Feature selection (DFS) was introduced to
overcome the limitations of previous feature selection models. In fact,
using a sparse regularized one-to-one linear layer on the input features
on deep neural networks, one can conveniently select a subset of features
for multiple classes and thus consider non-linearity via a deep
structure \cite{LiYifengDFS}. Convolutional Neural Networks (CNN) \cite{LecunCNN,
MartinezCNN}, a powerful deep learning algorithm for modelling
labelled sequential data was recently applied in DeepBind
\cite{AlipanahiDeepBind} in order to detect the Transcription Factor
Binding Site (TFBS).
Another step forward in distinguishing CRRs was accomplished by Wassermann
and his group in DECRES \cite{WassermannDECRES}. They achieved
excellent results using FANTOM promoters and enhancers (that provide the
largest experimentally defined collection of CRRs) with the ENCODE project
genome-wide feature data to
training supervised feed-forward neural networks in order to detect
regulatory regions. The data used contains histone modifications, TF binding, RNA transcripts, chromatin accessibility, and chromatin interactions.

%- previous works for MTL methods
In this work, we tried to make a step forward in predicting cis-regulatory
regions by applying Multi-Task Learning
(MTL) strategies to deep learning. Although MTL has little been used in bioinformatics,
especially in distinguish CRRs, a lot of MTL variants have been successfully applied to many
fields ranging from natural language processing \cite{CollobertWeston2008} to speech recognition \cite{Deng2013}, computer vision \cite{Girshick2015} and drug discovery \cite{Ramsundar2015}. An extensive overview of multi-task learning in deep neural network was written by Ruder \cite{Ruder2017}. Section~\ref{sec:MTLsection} provides a detailed explanation of Hard parameter sharing neural networks which is the MTL approach used in this work.